### PR TITLE
Refactor GUI-related code

### LINKED ISSUE #203: Refactor Analytics and GUI related code
The code for GUI, especially `MainWindow.java` has long methods that can be reorganized.

### PR DESCRIPTION
**Pull request by marcus-ny**
Fixes #203 

## Changelog

- Remove unused methods and code in `Analytics.java`
- Remove testcases related to removed methods
- Create `TabIndicator.java` to represent which tab to display currently
- Refactor tab switching logic to be handled by observable `TabIndicator` in `Model` class
- Minor tweaks to naming conventions (`loaneeInfoFlag` renamed to `isShowLoaneeInfo`)

### CODE REVIEW

File: src/main/java/seedu/address/ui/MainWindow.java [hunk 1]
```diff
@@ -171,70 +162,69 @@ private void initializePlaceholderSettings() {
...
-
-        this.isPersonTab.addListener((observable, oldValue, newValue) -> {
-            toggleTabs();
-        });
-        this.isLoansTab.addListener((observable, oldValue, newValue) -> {
-            toggleTabs();
-        });
-        this.isAnalyticsTab.addListener((observable, oldValue, newValue) -> {
+        this.tabIndicator.addListener((observable, oldValue, newValue) -> {
+            System.out.println("Value changed!" + newValue);
```
Comment (kjw142857) [inline src/main/java/seedu/address/ui/MainWindow.java:168]: This System.out.println statement probably needs to be removed!

**Review by kjw142857 — CHANGES_REQUESTED** [review]

File: src/main/java/seedu/address/ui/MainWindow.java (hunk 1 above)
Comment (marcus-ny) [reply src/main/java/seedu/address/ui/MainWindow.java:168]: Oops, will remove it now

**Review by Joseph31416 — APPROVED** [review]
Refactored code does not introduce regression and all features have the same expected behaviour, well done!

**Review by kjw142857 — APPROVED** [review]
LGTM too!

**Pull request merged**